vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/AbstractBrowserTestBase.java

```diff
@@ -336,7 +336,7 @@ protected boolean hasCssClass(WebElement element, String className) {
      * @param actualElement
      *            the actual element
      */
-    protected static void assertEquals(WebElement expectedElement,
+    protected static void assertEqual(WebElement expectedElement,
```

Comment:
Yes, the reason this method was renamed was due to its _collision_ with the JUnit assertion. Its name makes it impossible to statically import the JUnit assertion one. Preserving and deprecating would not resolve that issue. As this method is protected, is a temporary workaround, and may no longer be needed (per its Javadoc comment, below), it did not seem like an unreasonable change.
```
     * Can be removed if https://dev.vaadin.com/ticket/18484 is fixed.
```

Reply:
Not completely sure about renaming. For example, it will break `ShortcutsIT` in Flow repo.
Of course, it will not be a big effort to fix the test.
But if we rename it, this should be mentioned in the release notes.
@mshabarov what do you think?

Reply:
We're trying to avoid breaking changes as much as possible in minor releases, so I don't appreciate the immediate renaming.

The best way to resolve these collision in my opinion is to deprecate `AbstractBrowserTestBase::assertEquals` method and then at some point remove remove it in the possible new major. However, before we do this, we have to apply unwrapping from `AbstractBrowserTestBase::assertEquals` to the `TestBenchElement::equals` (recommended APi for comparing elements), as otherwise we can loose this functionality.
I'm talking about [this unwrapping](https://github.com/vaadin/testbench/blob/48b384c3be62e7c27bdaa7afb447172fbf051503/vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/AbstractBrowserTestBase.java#L343-L350).

Anyway, I think this change would be good to make, but it's not strictly related to the ElementQuery improvements, so I propose to make another PR or ticket for it.

Reply:
> However, before we do this, we have to apply unwrapping from AbstractBrowserTestBase::assertEquals to the TestBenchElement::equals (recommended APi for comparing elements), as otherwise we can loose this functionality.
> I'm talking about [this unwrapping](https://github.com/vaadin/testbench/blob/48b384c3be62e7c27bdaa7afb447172fbf051503/vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/AbstractBrowserTestBase.java#L343-L350).
> 

@mshabarov Yes, that unwrapping issue (https://dev.vaadin.com/ticket/18484) has long been resolved, so the `AbstractBrowserTestBase::assertEquals` method can be safely deprecated now and removed in the next major.

<img width="604" alt="image" src="https://github.com/vaadin/testbench/assets/62086947/5a31590c-0543-4c20-9f4d-98d537c4d279">

Reply:
So I will (sadly) revert this part of my change.